### ISSUE TITLE
Missing "longjmp" & "setjmp" functions in pre-built Newlib for ARCv3

### ISSUE THREAD

**Issue by abrodkin** [foss-for-synopsys-dwc-arc-processors/toolchain#512]
Seen with both `arc-2022.09-rc2` (see https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/tag/arc-2022.09-rc2) and built with "arc-gnu-toolchain" scripts, see https://github.com/foss-for-synopsys-dwc-arc-processors/arc-gnu-toolchain/actions/runs/4025073117 (note, here the same `arc-2022.09` branches are used as in the official release).

First observer on building of some test applications:
```
.../arc-2022.09-rc2/arc64_elf_linux/bin/../lib/gcc/arc64-snps-elf/12.2.1/../../../../arc64-snps-elf/bin/ld: .../tests/./common/thal.c:574: undefined reference to `longjmp'
.../arc-2022.09-rc2/arc64_elf_linux/bin/../lib/gcc/arc64-snps-elf/12.2.1/../../../../arc64-snps-elf/bin/ld: /tmp/ccmikchj.ltrans0.ltrans.o: in function `main':
.../tests/./common/thal.c:682: undefined reference to `setjmp'
```

And in fact:
```
$ ./bin/arc64-elf-nm arc64-elf/lib/libc.a | grep longjmp
$ ./bin/arc64-elf-nm arc64-elf/lib/libc.a | grep setjmp
```

While in case of ARCompact/ARCv2 toolchain:
```
$ ./bin/arc-elf32-nm arc-elf32/lib/libc.a | grep longjmp
00000068 T longjmp

$ ./bin/arc-elf32-nm arc-elf32/lib/libc.a | grep setjmp
libc_a-setjmp.o:
```

In `arc-2021.09` that problem is not observed as well:
```
$ ./bin/arc64-elf-nm arc64-snps-elf/lib/libc.a | grep setjmp
lib_a-setjmp.o:
0000000000000000 T setjmp

$ ./bin/arc64-elf-nm arc64-snps-elf/lib/libc.a | grep longjmp
0000000000000050 T longjmp
```

**Comment by shahab-vahedi** [comment]
The [`arc-2022.09`](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/tree/arc-2022.09) branch has been cut out again. This should resolve the matter. Please try again and update this issue.

**Comment by shahab-vahedi** [comment]
Moved back to previous state (commit `3600e6b88`), till we decide what to do with this.

**Comment by shahab-vahedi** [comment]
@claziss and I investigated this issue and came up with the following patches on top of [arc-2022.09](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/tree/arc-2022.09) branch:

1. [arc64: hand craft the build infra (4e191d542)](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/commit/4e191d542)
2. [arc64: regenerate the build infra (f3c4aadca)](https://github.com/foss-for-synopsys-dwc-arc-processors/newlib/commit/f3c4aadca)


Please verify if this resolves the issue on your end and update here accordingly.

**Comment by shahab-vahedi** [comment]
Any update?

**Comment by abrodkin** [comment]
Fixed in the final release 2022.09, see https://github.com/foss-for-synopsys-dwc-arc-processors/toolchain/releases/tag/arc-2022.09-release
